File: android/src/main/kotlin/com/tsitser/background_http_plugin/domain/usecase/GetRequestStatusUseCase.kt
package com.tsitser.background_http_plugin.domain.usecase

import com.tsitser.background_http_plugin.domain.entity.TaskInfo
import com.tsitser.background_http_plugin.domain.repository.TaskRepository

/**
 * Use case для получения статуса задачи
 */
class GetRequestStatusUseCase(
    private val repository: TaskRepository
) {
    suspend operator fun invoke(requestId: String): TaskInfo? {
        return repository.getTaskInfo(requestId)
    }
}

File: ios/Classes/Domain/UseCase/GetRequestStatusUseCase.swift
import Foundation

/// Use case for getting task status
struct GetRequestStatusUseCase {
    private let repository: TaskRepository
    
    init(repository: TaskRepository) {
        self.repository = repository
    }
    
    func execute(requestId: String) async throws -> TaskInfo? {
        return try await repository.getTaskInfo(requestId: requestId)
    }
}

